```
10:46:40 [WARNINGS] Parsing warnings in console log with parser CMake
10:46:41 <Git Blamer> Using GitBlamer to create author and commit information for all warnings.
10:46:41 <Git Blamer> GIT_COMMIT=66fb2412443071598201b986de8db2e6ccfb46ca, workspace=/home/jenkins-agent/workspace/Mpr__perception_pcl__ubuntu_bionic_amd64/catkin_workspace/src/perception_pcl
10:46:41 [WARNINGS] Parsing warnings in console log with parser GNU C Compiler 4 (gcc)
10:46:41 <Git Blamer> Using GitBlamer to create author and commit information for all warnings.
10:46:41 <Git Blamer> GIT_COMMIT=66fb2412443071598201b986de8db2e6ccfb46ca, workspace=/home/jenkins-agent/workspace/Mpr__perception_pcl__ubuntu_bionic_amd64/catkin_workspace/src/perception_pcl
10:46:41 <Git Blamer> Skipping non-workspace file /usr/include/pcl-1.8/pcl/sample_consensus/model_types.h (workspace = /home/jenkins-agent/workspace/Mpr__perception_pcl__ubuntu_bionic_amd64/catkin_workspace/src/perception_pcl, absolute = /usr/include/pcl-1.8/pcl/sample_consensus/model_types.h.
10:46:41  > git rev-parse 66fb2412443071598201b986de8db2e6ccfb46ca^{commit} # timeout=10
10:46:41 <Git Blamer> Skipping file /usr/include/pcl-1.8/pcl/sample_consensus/model_types.h, no result found.
10:46:41 [WARNINGS] Computing warning deltas based on reference build #20
10:46:41 [WARNINGS] Ignore new warnings since this is the first valid build
10:46:41 [WARNINGS] Plug-in Result: Unstable - <a href="warningsResult">1 warning</a> exceeds the threshold of 0 by 1
10:46:41 Build step 'Scan for compiler warnings' changed build result to UNSTABLE
10:46:41 [xUnit] [INFO] - Starting to record.
10:46:41 [xUnit] [INFO] - Processing GoogleTest-1.6
10:46:41 [xUnit] [INFO] - [GoogleTest-1.6] - 13 test report file(s) were found with the pattern 'catkin_workspace/test_results/**/*.xml' relative to '/home/jenkins-agent/workspace/Mpr__perception_pcl__ubuntu_bionic_amd64' for the testing framework 'GoogleTest-1.6'.
10:46:41 [xUnit] [INFO] - Check 'Failed Tests' threshold.
10:46:41 [xUnit] [INFO] - Check 'Skipped Tests' threshold.
10:46:41 [xUnit] [INFO] - Setting the build status to UNSTABLE
10:46:41 [xUnit] [INFO] - Stopping recording.
10:46:41 Setting status of ec9c00ff06db1796a19a1d11a8004975963f015c to FAILURE with url http://build.ros.org/job/Mpr__perception_pcl__ubuntu_bionic_amd64/22/ and message: 'Build finished. '
10:46:41 Using context: Mpr__perception_pcl__ubuntu_bionic_amd64
10:46:41 Could not update commit status of the Pull Request on GitHub.
10:46:41 org.kohsuke.github.HttpException: Server returned HTTP response code: 201, message: 'Created' for URL: https://api.github.com/repos/ros-perception/perception_pcl/statuses/ec9c00ff06db1796a19a1d11a8004975963f015c
10:46:41 	at org.kohsuke.github.Requester.parse(Requester.java:622)
10:46:41 	at org.kohsuke.github.Requester.parse(Requester.java:584)
10:46:41 	at org.kohsuke.github.Requester._to(Requester.java:264)
10:46:41 	at org.kohsuke.github.Requester.to(Requester.java:226)
10:46:41 	at org.kohsuke.github.GHRepository.createCommitStatus(GHRepository.java:1000)
10:46:41 	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.createCommitStatus(GhprbSimpleStatus.java:272)
10:46:41 	at org.jenkinsci.plugins.ghprb.extensions.status.GhprbSimpleStatus.onBuildComplete(GhprbSimpleStatus.java:231)
10:46:41 	at org.jenkinsci.plugins.ghprb.GhprbBuilds.onCompleted(GhprbBuilds.java:192)
10:46:41 	at org.jenkinsci.plugins.ghprb.GhprbBuildListener.onCompleted(GhprbBuildListener.java:26)
10:46:41 	at hudson.model.listeners.RunListener.fireCompleted(RunListener.java:211)
10:46:41 	at hudson.model.Run.execute(Run.java:1780)
10:46:41 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
10:46:41 	at hudson.model.ResourceController.execute(ResourceController.java:97)
10:46:41 	at hudson.model.Executor.run(Executor.java:405)
10:46:41 Caused by: java.io.IOException: Failed to deserialize {"url":"https://api.github.com/repos/ros-perception/perception_pcl/statuses/ec9c00ff06db1796a19a1d11a8004975963f015c","id":4904279101,"state":"failure","description":"Build finished. ","target_url":"http://build.ros.org/job/Mpr__perception_pcl__ubuntu_bionic_amd64/22/","context":"Mpr__perception_pcl__ubuntu_bionic_amd64","created_at":"2018-05-02T17:46:41Z","updated_at":"2018-05-02T17:46:41Z","creator":{"login":"ros-pull-request-builder","id":7094544,"avatar_url":"https://avatars0.githubusercontent.com/u/7094544?v=4","gravatar_id":"","url":"https://api.github.com/users/ros-pull-request-builder","html_url":"https://github.com/ros-pull-request-builder","followers_url":"https://api.github.com/users/ros-pull-request-builder/followers","following_url":"https://api.github.com/users/ros-pull-request-builder/following{/other_user}","gists_url":"https://api.github.com/users/ros-pull-request-builder/gists{/gist_id}","starred_url":"https://api.github.com/users/ros-pull-request-builder/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/ros-pull-request-builder/subscriptions","organizations_url":"https://api.github.com/users/ros-pull-request-builder/orgs","repos_url":"https://api.github.com/users/ros-pull-request-builder/repos","events_url":"https://api.github.com/users/ros-pull-request-builder/events{/privacy}","received_events_url":"https://api.github.com/users/ros-pull-request-builder/received_events","type":"User","site_admin":false}}
10:46:41 	at org.kohsuke.github.Requester.parse(Requester.java:608)
10:46:41 	... 13 more
10:46:41 Caused by: com.fasterxml.jackson.databind.JsonMappingException: Numeric value (4904279101) out of range of int
10:46:41  at [Source: {"url":"https://api.github.com/repos/ros-perception/perception_pcl/statuses/ec9c00ff06db1796a19a1d11a8004975963f015c","id":4904279101,"state":"failure","description":"Build finished. ","target_url":"http://build.ros.org/job/Mpr__perception_pcl__ubuntu_bionic_amd64/22/","context":"Mpr__perception_pcl__ubuntu_bionic_amd64","created_at":"2018-05-02T17:46:41Z","updated_at":"2018-05-02T17:46:41Z","creator":{"login":"ros-pull-request-builder","id":7094544,"avatar_url":"https://avatars0.githubusercontent.com/u/7094544?v=4","gravatar_id":"","url":"https://api.github.com/users/ros-pull-request-builder","html_url":"https://github.com/ros-pull-request-builder","followers_url":"https://api.github.com/users/ros-pull-request-builder/followers","following_url":"https://api.github.com/users/ros-pull-request-builder/following{/other_user}","gists_url":"https://api.github.com/users/ros-pull-request-builder/gists{/gist_id}","starred_url":"https://api.github.com/users/ros-pull-request-builder/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/ros-pull-request-builder/subscriptions","organizations_url":"https://api.github.com/users/ros-pull-request-builder/orgs","repos_url":"https://api.github.com/users/ros-pull-request-builder/repos","events_url":"https://api.github.com/users/ros-pull-request-builder/events{/privacy}","received_events_url":"https://api.github.com/users/ros-pull-request-builder/received_events","type":"User","site_admin":false}}; line: 1, column: 134]
10:46:41  at [Source: {"url":"https://api.github.com/repos/ros-perception/perception_pcl/statuses/ec9c00ff06db1796a19a1d11a8004975963f015c","id":4904279101,"state":"failure","description":"Build finished. ","target_url":"http://build.ros.org/job/Mpr__perception_pcl__ubuntu_bionic_amd64/22/","context":"Mpr__perception_pcl__ubuntu_bionic_amd64","created_at":"2018-05-02T17:46:41Z","updated_at":"2018-05-02T17:46:41Z","creator":{"login":"ros-pull-request-builder","id":7094544,"avatar_url":"https://avatars0.githubusercontent.com/u/7094544?v=4","gravatar_id":"","url":"https://api.github.com/users/ros-pull-request-builder","html_url":"https://github.com/ros-pull-request-builder","followers_url":"https://api.github.com/users/ros-pull-request-builder/followers","following_url":"https://api.github.com/users/ros-pull-request-builder/following{/other_user}","gists_url":"https://api.github.com/users/ros-pull-request-builder/gists{/gist_id}","starred_url":"https://api.github.com/users/ros-pull-request-builder/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/ros-pull-request-builder/subscriptions","organizations_url":"https://api.github.com/users/ros-pull-request-builder/orgs","repos_url":"https://api.github.com/users/ros-pull-request-builder/repos","events_url":"https://api.github.com/users/ros-pull-request-builder/events{/privacy}","received_events_url":"https://api.github.com/users/ros-pull-request-builder/received_events","type":"User","site_admin":false}}; line: 1, column: 124] (through reference chain: org.kohsuke.github.GHCommitStatus["id"])
10:46:41 	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:339)
10:46:41 	at com.fasterxml.jackson.databind.JsonMappingException.wrapWithPath(JsonMappingException.java:299)
10:46:41 	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.wrapAndThrow(BeanDeserializerBase.java:1511)
10:46:41 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:262)
10:46:41 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:125)
10:46:41 	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3789)
10:46:41 	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2779)
10:46:41 	at org.kohsuke.github.Requester.parse(Requester.java:606)
10:46:41 	... 13 more
10:46:41 Caused by: com.fasterxml.jackson.core.JsonParseException: Numeric value (4904279101) out of range of int
10:46:41  at [Source: {"url":"https://api.github.com/repos/ros-perception/perception_pcl/statuses/ec9c00ff06db1796a19a1d11a8004975963f015c","id":4904279101,"state":"failure","description":"Build finished. ","target_url":"http://build.ros.org/job/Mpr__perception_pcl__ubuntu_bionic_amd64/22/","context":"Mpr__perception_pcl__ubuntu_bionic_amd64","created_at":"2018-05-02T17:46:41Z","updated_at":"2018-05-02T17:46:41Z","creator":{"login":"ros-pull-request-builder","id":7094544,"avatar_url":"https://avatars0.githubusercontent.com/u/7094544?v=4","gravatar_id":"","url":"https://api.github.com/users/ros-pull-request-builder","html_url":"https://github.com/ros-pull-request-builder","followers_url":"https://api.github.com/users/ros-pull-request-builder/followers","following_url":"https://api.github.com/users/ros-pull-request-builder/following{/other_user}","gists_url":"https://api.github.com/users/ros-pull-request-builder/gists{/gist_id}","starred_url":"https://api.github.com/users/ros-pull-request-builder/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/ros-pull-request-builder/subscriptions","organizations_url":"https://api.github.com/users/ros-pull-request-builder/orgs","repos_url":"https://api.github.com/users/ros-pull-request-builder/repos","events_url":"https://api.github.com/users/ros-pull-request-builder/events{/privacy}","received_events_url":"https://api.github.com/users/ros-pull-request-builder/received_events","type":"User","site_admin":false}}; line: 1, column: 134]
10:46:41 	at com.fasterxml.jackson.core.JsonParser._constructError(JsonParser.java:1586)
10:46:41 	at com.fasterxml.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:521)
10:46:41 	at com.fasterxml.jackson.core.base.ParserBase.convertNumberToInt(ParserBase.java:938)
10:46:41 	at com.fasterxml.jackson.core.base.ParserBase._parseIntValue(ParserBase.java:876)
10:46:41 	at com.fasterxml.jackson.core.base.ParserBase.getIntValue(ParserBase.java:704)
10:46:41 	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:313)
10:46:41 	at com.fasterxml.jackson.databind.deser.std.NumberDeserializers$IntegerDeserializer.deserialize(NumberDeserializers.java:293)
10:46:41 	at com.fasterxml.jackson.databind.deser.SettableBeanProperty.deserialize(SettableBeanProperty.java:490)
10:46:41 	at com.fasterxml.jackson.databind.deser.impl.FieldProperty.deserializeAndSet(FieldProperty.java:101)
10:46:41 	at com.fasterxml.jackson.databind.deser.BeanDeserializer.vanillaDeserialize(BeanDeserializer.java:260)
10:46:41 	... 17 more
10:46:41 Finished: UNSTABLE
```
http://build.ros.org/job/Mpr__perception_pcl__ubuntu_bionic_amd64/22/console

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,12 +5,17 @@ services:
   - docker
 cache: ccache
 env:
-  global:
-    - ROS_DISTRO=melodic
   matrix:
-    - DOCKER_IMAGE=debian:stretch
-    - DOCKER_IMAGE=ubuntu:artful
-    - DOCKER_IMAGE=ubuntu:bionic
+    # Test the target distro.
+    - ROS_DISTRO=melodic DOCKER_IMAGE=debian:stretch
+    - ROS_DISTRO=melodic DOCKER_IMAGE=ubuntu:artful
+    - ROS_DISTRO=melodic DOCKER_IMAGE=ubuntu:bionic
+    # To test backward compatibility for users who build from source.
+    - ROS_DISTRO=lunar DOCKER_IMAGE=debian:stretch
+    - ROS_DISTRO=lunar DOCKER_IMAGE=ubuntu:xenial
+    - ROS_DISTRO=kinetic DOCKER_IMAGE=debian:jessie
+    - ROS_DISTRO=kinetic DOCKER_IMAGE=ubuntu:xenial
+    - ROS_DISTRO=indigo DOCKER_IMAGE=ubuntu:trusty
 # Install system dependencies, namely ROS.
 before_install:
   # Define some config vars.

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,10 +6,11 @@ services:
 cache: ccache
 env:
   global:
-    - ROS_DISTRO=lunar
+    - ROS_DISTRO=melodic
   matrix:
     - DOCKER_IMAGE=debian:stretch
-    - DOCKER_IMAGE=ubuntu:xenial
+    - DOCKER_IMAGE=ubuntu:artful
+    - DOCKER_IMAGE=ubuntu:bionic
 # Install system dependencies, namely ROS.
 before_install:
   # Define some config vars.

--- a/pcl_conversions/CMakeLists.txt
+++ b/pcl_conversions/CMakeLists.txt
@@ -13,7 +13,12 @@ find_package(Eigen3 REQUIRED)
 # configure properly.  Here we remove those bogus entries so that downstream
 # consumers of this package succeed.
 if(NOT "${PCL_INCLUDE_DIRS}" STREQUAL "")
-  list(FILTER PCL_INCLUDE_DIRS EXCLUDE REGEX "/usr/include/.*-linux-gnu/freetype2")
+  foreach(item ${PCL_INCLUDE_DIRS})
+    string(REGEX MATCH "/usr/include/.*-linux-gnu/freetype2" item ${item})
+    if(item)
+      list(REMOVE_ITEM PCL_INCLUDE_DIRS ${item})
+    endif()
+  endforeach()
 endif()
 
 catkin_package(

